pkg/controller/servicecidrs/servicecidrs_controller.go

```diff
@@ -302,6 +302,14 @@ func (c *Controller) sync(ctx context.Context, key string) error {
 			// update the status to indicate why the ServiceCIDR can not be deleted,
 			// it will be reevaludated by an event on any ServiceCIDR or IPAddress related object
 			// that may remove this condition.
+			for _, condition := range cidr.Status.Conditions {
+				if condition.Type == networkingapiv1.ServiceCIDRConditionReady {
+					if condition.Status == metav1.ConditionFalse &&
+						condition.Reason == networkingapiv1.ServiceCIDRReasonTerminating {
+						return nil
```

Comment:
I'm pretty sure there's a helper method for "find a condition that looks like this one", and maybe even "add this condition only if it doesn't exist"

Reply:
added one more specific for this controller